src/dosext/mfs/mfs.c

```diff
@@ -372,6 +372,58 @@ static uint16_t GetDataSegment(void)
   return ds;
 }
 
+/*****************************
+ * GetCurrentDriveInDOS
+ * on entry:
+ *   drv: For the result (0=A, 1=B etc)
+ * on exit:
+ *   Returns 1 on success, 0 on fail
+ * notes:
+ *   This function can be used only whilst InDOS, in essence that means from
+ *   redirector int2f/11 function. It relies on being able to run int2f/12
+ *   functions which need to use the DOS stack. Outside of DOS use int21/19
+ *   which is much easier in any case
+ *****************************/
+static int GetCurrentDriveInDOS(uint8_t *drv)
+{
+  struct vm86_regs saved_regs = REGS;
+  char *src, *dst;
+  uint8_t dd;
+
+  /* Ask DOS to canonicalize '\' which should give us 'X:\' back */
+
+  /* Use 2 bytes from the DOS stack */
+  _SP -= 2;
+  src = MK_FP32(_SS, _SP);
+  memcpy(src, "\\", 2);
+  _DS = _SS;			// DS:SI -> file name to be fully qualified
+  _SI = _SP;
+
+  /* DOS buffer we know is not in use */
```

Comment:
Why not lowmem_*()?

Reply:
Is it bad to use the DOS stack? Can switch to using lowmem, but wasn't sure if okay inside DOS? As an alternative I can
- place it at the end of LFN_short_name (not sure about src/dst overlapping)
- steal a couple more bytes from BIOS

Reply:
> Is it bad to use the DOS stack?

Absolutely, it is internal to dosemu.

> steal a couple more bytes from BIOS

That's what lowmem_alloc() exactly does.